### PR TITLE
[New Rule] Potential PowerShell Obfuscated Script

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/3293

## Summary

Identifies scripts that contain patterns and known methods that obfuscate PowerShell code. Attackers can use obfuscation techniques to bypass PowerShell security protections such as Antimalware Scan Interface (AMSI).

Some techniques, such as special chars & whitespace encoding and more generic reorder and concatenation, aren't covered because of language (KQL) limitations.

LS24 Stack:

image

### CODE REVIEW

**Review by terrancedejesus — APPROVED** [review]
Looks good! Reviewed cmdlets and setup guide. Checked index patterns and tags.

File: rules/windows/defense_evasion_posh_obfuscation.toml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+description = """
+Identifies scripts that contain patterns and known methods that obfuscate PowerShell code. Attackers can use obfuscation
+techniques to bypass PowerShell security protections such as Antimalware Scan Interface (AMSI).
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-windows.powershell*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Potential PowerShell Obfuscated Script"
+reference = ["https://github.com/danielbohannon/Invoke-Obfuscation"]
```
Comment (Mikaayenson) [inline rules/windows/defense_evasion_posh_obfuscation.toml:18]:
```suggestion
references = ["https://github.com/danielbohannon/Invoke-Obfuscation"]
```

Should save you time trying to figure out the unit test error.

**Review by Mikaayenson — COMMENTED** [review]
I couldn't tell based on the summary, does this also cover behaviors of [Invoke-Obfuscation.ps1](https://github.com/danielbohannon/Invoke-Obfuscation/blob/master/Invoke-Obfuscation.ps1)?

**Review by Aegrah — APPROVED** [review]
The query looks quite generic to me in some cases, potentially leading to a lot of hits. Maybe it could make sense to add it as a new_terms rule type, having the script name as the unique value, and take advantage of multiple hosts to reduce FPs. Up to you. Otherwise, LGTM

**Comment by w0rk3r** [comment]
> I couldn't tell based on the summary, does this also cover behaviors of [Invoke-Obfuscation.ps1](https://github.com/danielbohannon/Invoke-Obfuscation/blob/master/Invoke-Obfuscation.ps1)?

Yup

> The query looks quite generic to me in some cases, potentially leading to a lot of hits

I've excluded some patterns that were matching in ansible scripts, powershell modules and some other legitimate stuff, I think it wouldn't be that noisy

> Maybe it could make sense to add it as a new_terms rule type, having the script name as the unique value, and take advantage of multiple hosts to reduce FPs

Most of the malicious ones will not be backed by a file, I'll push this one as is and then I can evaluate conversion to new terms if it is too noisy.

**Pull request merged**